windows/src/engine/keyman32/K32_load.cpp

```diff
@@ -75,12 +75,55 @@ BOOL GetKeyboardFileName(LPSTR kbname, LPSTR buf, int nbuf)
   return n;
 }
 
+BOOL LoadlpKeyboardCore(int i)
+{
+  SendDebugMessageFormat(0, sdmAIDefault, 0, "LoadlpKeyboardCore: Enter ---");
+  PKEYMAN64THREADDATA _td = ThreadGlobals();
+  if (!_td) return FALSE;
+  if (_td->lpKeyboards[i].coreKeyboard) return TRUE;
+  if (_td->lpActiveKeyboard == &_td->lpKeyboards[i]) _td->lpActiveKeyboard = NULL;  // I822 TSF not working
+
+  if (_td->lpKeyboards[i].lpActiveKBState) {
+    km_kbp_state_dispose(_td->lpKeyboards[i].lpActiveKBState);
+    _td->lpKeyboards[i].lpActiveKBState = NULL;
+  }
+
+  char buf[256];
+  if (!GetKeyboardFileName(_td->lpKeyboards[i].Name, buf, 255)) return FALSE;
+  PWCHAR keyboardPath = strtowstr(buf);
+  km_kbp_status_codes err_code = (km_kbp_status_codes)km_kbp_keyboard_load(keyboardPath, &_td->lpKeyboards[i].coreKeyboard);
```

Comment:
Why do we need to cast function result here? This is not ideal...

Reply:
I will check this again it gave a compiler warning about an implicit cast. However worse than that it was failing the if when comparison on line 95 which surprised me when using windbg to step through. I will check again with straight logging. 
I have a question though why have public enum available in keyboardprocessor.h and instead for returning that type return a uint32_t?